File: musicShopProject/Model/Orders/OrderItem.cs
using musicShopProject.Model.Products;
using musicShopProject.Service.Orders.Repository.Model;

namespace musicShopProject.Model.Orders;

public partial class OrderItem
{
    public Guid? Id { get; }
    public Guid? OrderId { get; }
    public Product? Product { get; }
    public Decimal? ProductPrice { get; }
    public Guid? ProductCategoryId { get; set; }

    public OrderItem(OrderItemDB db, Product product)
    {
        Id = db.Id;
        OrderId = db.OrderId;
        Product = product;
        ProductPrice = db.ProductPrice;
        ProductCategoryId = db.ProductCategoryId;
    }
}

File: musicShopProject/tentashop/src/domain/orders/orderItem.ts
import { mapToProduct, Product } from "../products/product";

export class OrderItem {
    constructor(
        public id: string,
        public orderId: string,
        public product: Product,
        public productPrice: number,
        public productCategoryId: string
    ) { }
}

export function mapToOrderItem(data: any): OrderItem {
    const product = mapToProduct(data.product);

    return new OrderItem(data.id, data.orderId, product, data.productPrice, data.productCategoryId);
}

export function mapToOrderItems(data: any[]): OrderItem[] {
    return data.map(mapToOrderItem);
}
